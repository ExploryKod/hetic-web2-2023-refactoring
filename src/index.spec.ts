import runGoldenMaster from "jest-golden-master";
import { Game } from ".";

test("A player wins by providing correct answers each time", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    game.roll(1);
    game.wasCorrectlyAnswered();

    // Mathieu
    game.roll(1);
    game.wasCorrectlyAnswered();
  });
});
