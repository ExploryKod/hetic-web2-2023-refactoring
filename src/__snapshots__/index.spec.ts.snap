// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A player wins by providing correct answers each time 1`] = `
[
  "New player added: Mathieu; their place is 0 and they have 0 coins. They are NOT in the penalty box.",
  "New player added: Thomas; their place is 0 and they have 0 coins. They are NOT in the penalty box.",
  "Mathieu rolled a 1.",
  "Science Question 0",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 1 points.",
  "The new current player is Thomas.",
  "Thomas rolled a 1.",
  "Science Question 1",
  "Thomas provided the correct answer.",
  "Thomas earned a coin and consequently has 1 points.",
  "The new current player is Mathieu.",
  "Mathieu rolled a 1.",
  "Sports Question 0",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 2 points.",
  "The new current player is Thomas.",
  "Thomas rolled a 1.",
  "Sports Question 1",
  "Thomas provided the correct answer.",
  "Thomas earned a coin and consequently has 2 points.",
  "The new current player is Mathieu.",
  "Mathieu rolled a 1.",
  "Rock Question 0",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 3 points.",
  "The new current player is Thomas.",
  "Thomas rolled a 1.",
  "Rock Question 1",
  "Thomas provided the correct answer.",
  "Thomas earned a coin and consequently has 3 points.",
  "The new current player is Mathieu.",
  "Mathieu rolled a 1.",
  "Pop Question 0",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 4 points.",
  "The new current player is Thomas.",
  "Thomas rolled a 1.",
  "Pop Question 1",
  "Thomas provided the correct answer.",
  "Thomas earned a coin and consequently has 4 points.",
  "The new current player is Mathieu.",
  "Mathieu rolled a 1.",
  "Science Question 2",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 5 points.",
  "The new current player is Thomas.",
  "Thomas rolled a 1.",
  "Science Question 3",
  "Thomas provided the correct answer.",
  "Thomas earned a coin and consequently has 5 points.",
  "The new current player is Mathieu.",
  "Mathieu rolled a 1.",
  "Sports Question 2",
  "Mathieu provided the correct answer.",
  "Mathieu earned a coin and consequently has 6 points.",
  "Mathieu won the game.",
  "The new current player is Thomas.",
]
`;
